.github/workflows/check_new_kernel_configs.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Script to check for new configs which should be in required kernel configs json
+# Usage: python3 check_new_kernel_configs.py --required_configs <path to json of required configs> --config_str <string of diff for config file>
+
+import json
+import argparse
+import sys
+import re
+
+# find the lines in the diff that contain +/- CONFIG
+# ignore the lines that contain @
+def find_matching_lines(input_string):
+    pattern = r'(?!.*@)[+-]\s*.*CONFIG.*'
+    matching_lines = re.findall(pattern, input_string)
+
+    return matching_lines
+
+# parse diff for new kernel configs
+# check if they are in required configs
+def find_missing_words(json_file, string):
```

Comment:
Could we use a different variable name than `string`? It's a keyword in so many languages, that this can be a bit confusing. Also, it this name doesn't really provide a lot of information to the reader.